File: glowxq-api/business/business-system/src/main/java/com/glowxq/system/meta/enums/MenuIconType.java
package com.glowxq.system.meta.enums;

import com.glowxq.core.common.enums.base.BaseType;
import com.glowxq.system.meta.base.MenuIconEnum;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author glowxq
 * @version 1.0
 * @date 2025/1/20
 */
@AllArgsConstructor
@Getter
public enum MenuIconType implements BaseType, MenuIconEnum {

    Icon("Icon", "Icon"),
    Image("Image", "图片"),

    ;

    /**
     * code
     */
    private final String code;

    /**
     * 最近类型
     */
    private final String name;

    /**
     * 根据code获取枚举
     *
     * @param code
     * @return
     */
    public static MenuIconType matchCode(String code) {
        for (MenuIconType pushStatus : MenuIconType.values()) {
            if (pushStatus.getCode().equals(code)) {
                return pushStatus;
            }
        }
        return null;
    }
}

File: oj-web/src/enums/system/meta/MenuIconType.ts
import type { BaseEnum } from '@/enums/base';

// 菜单图标类型枚举接口
interface MenuIconTypeEnum extends BaseEnum {
  tooltip?: string;
  text?: string;
}

/**
 * 菜单图标类型枚举
 * 用于标识不同的菜单图标类型
 */
export const MenuIconType: { [key: string]: MenuIconTypeEnum } = {
  /**
   * 图标
   */
  ICON: {
    code: 'Icon',
    name: 'Icon',
    tooltip: '图标类型',
    text: '图标类型'
  },

  /**
   * 图片
   */
  IMAGE: {
    code: 'Image',
    name: '图片',
    tooltip: '图片类型',
    text: '图片类型'
  }
};

/**
 * 根据code获取枚举
 * @param code
 * @returns 枚举值或null
 */
export function matchCode(code: string): MenuIconTypeEnum | null {
  return Object.values(MenuIconType).find(item => item.code === code) || null;
} 